packages/artifact/src/internal/upload/blob-upload.ts

```diff
@@ -37,8 +39,19 @@ export async function uploadZipToBlobStorage(
   const uploadCallback = (progress: TransferProgressEvent): void => {
     core.info(`Uploaded bytes ${progress.loadedBytes}`)
     uploadByteCount = progress.loadedBytes
+    progressTimeout(timeoutDuration)
   }
 
+  // Timeout if the upload stalls
+  const progressTimeout = (timeout: number): NodeJS.Timeout =>
+    setTimeout(() => {
```

Comment:
let's set this to a variable scoped outside of this like `timer` and use that to clear the interval